includes/compat/class-base.php

```diff
@@ -0,0 +1,53 @@
+<?php
+/**
+ * A Base Backwards Compatibility Class.
+ *
+ * @package     EDD
+ * @subpackage  Compat
+ * @copyright   Copyright (c) 2018, Easy Digital Downloads, LLC
+ * @license     http://opensource.org/licenses/gpl-2.0.php GNU Public License
+ * @since       3.0.0
+ */
+namespace EDD\Compat;
+
+// Exit if accessed directly
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Base class.
+ *
+ * @since 3.0.0
+ */
+abstract class Base {
```

Comment:
Can we add an abstract `hooks()` method to `Base` to enforce the contract for subclasses?